spark/src/main/scala/io/delta/sharing/spark/DeltaSharingDataSource.scala

```diff
@@ -28,29 +28,15 @@ import org.apache.spark.sql.types.StructType
 import org.apache.spark.sql.util.CaseInsensitiveStringMap
 
 /** A DataSource V1 for integrating Delta into Spark SQL batch APIs. */
-private[sharing] class DeltaSharingDataSource extends TableProvider with RelationProvider
-  with DataSourceRegister {
-
-  override def inferSchema(options: CaseInsensitiveStringMap): StructType = new StructType()
-
-  override def getTable(
-      schema: StructType,
-      partitioning: Array[Transform],
-      properties: java.util.Map[String, String]): Table = {
-    // Return a Table with no capabilities so we fall back to the v1 path.
-    new Table {
-      override def name(): String = s"V1FallbackTable"
-      override def schema(): StructType = new StructType()
-      override def capabilities(): java.util.Set[TableCapability] = Collections.emptySet()
-    }
-  }
+private[sharing] class DeltaSharingDataSource extends RelationProvider with DataSourceRegister {
```

Comment:
`TableProvider` was added to support advanced features such as time travel. Since we don't support these features today, I just remove it so that we will block managed tables properly.

Reply:
To make sure I'm understanding correctly, this blocks managed tables because we will no longer be able to fall back to the v1 path, so only paths in the form of `profile_file + "#<share-name>.<schema-name>.<table-name>"` can be created?

Reply:
@FX196 that's my understanding. Then what happens if we need to add `TableProvider` back to support advanced features?

Reply:
Removing TableProvider will make Spark to call `createRelation` below and verify the path. 

> Then what happens if we need to add TableProvider back to support advanced features?

We can add it back in future if we need. Since we have a test for blocking managed tables, adding it back in future would need to fix the test, such as load the table in `getTable` I removed in this PR.

Reply:
Removing TableProvider will make Spark to call `createRelation` below and verify the path. 

> Then what happens if we need to add TableProvider back to support advanced features?

We can add it back in future if we need. Since we have a test for blocking managed tables, adding it back in future would need to fix the test, such as load the table in `getTable` I removed in this PR.

Reply:
Makes sense. If we want to add it back in the future we can check the path in `getTable`?